### PR TITLE
Navigator part 2: URO-90 URO-109 URO-110 URO-112 URO-121

### PR DESCRIPTION
**Pull request by dakotablair**
- [x] [URO-90](https://kbase-jira.atlassian.net/browse/URO-90) Handle navigation and global state of navigator Feature
  - [x] refreshes state based on navigation actions from category tabs (e.g. /shared, /pubilc)
  - [x] refreshes state based on sort and search component query params
  - [x] Navigates upa url when user clicks on narrative from within NarrativeList
  - [x] Ensures the correct narrative is selected regardless of navigation actions
- [x] [URO-109](https://kbase-jira.atlassian.net/browse/URO-109) Create Search Bar component for Navigator
  - [x] Takes text as input
  - [x] Adds “search” queryParameter to url and navigates upon input change event with user input value
  - [x] removes search queryParam from url when user input is cleared
  - [x] retains input value across navigations, takes “value” as prop from  “search” queryParam
  - [x] Factor component into its own module
  - [x] Write tests with 100% coverage
- [x] [URO-110](https://kbase-jira.atlassian.net/browse/URO-110) Create Sort dropdown component for Navigator
  - [x] Uses dropdown with static values
  - [x] dropdown contains the specified values
  - [x] Default selection is determined by the “sort” search parameter.
  - [x] Factor component into its own module
  - [x] Write tests with 100% coverage
- [x] [URO-112](https://kbase-jira.atlassian.net/browse/URO-112) Create Refresh button
  - [x] Button takes prop that is a callback to the function used to fetch search results with current queryParams
  - [x] Clicking will cause the top level narrative component to reload and rerender search results accordingly
  - [x] Factor component into its own module
  - [x] Write tests with 100% coverage
- [x] [URO-121](https://kbase-jira.atlassian.net/browse/URO-121) Create Navigator Narrative Version Selection component
  - [x] Displays current version selected
  - [x] Provides dropdown to select from all other versions.
  - [x] What is selected is determined by the URL and when a new version is selected the URL changes accordingly, preserving
    - [x] search and sort query parameters
    - [x] category designation

### CODE REVIEW

File: src/features/navigator/Navigator.test.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+test('Navigator renders', () => {
+  const { container } = render(
+    <Provider store={createTestStore()}>
+      <Router>
+        <Navigator />
+      </Router>
+    </Provider>
+  );
+  expect(container).toBeTruthy();
+  screen.debug();
```
Comment (dauglyon) [inline src/features/navigator/Navigator.test.tsx:16]: ?

File: src/features/navigator/navigatorSlice.ts [hunk 2]
```diff
@@ -20,18 +29,50 @@ export const navigatorSlice = createSlice({
...
+      state.fresh = action.payload;
+    },
+    setNarratives: (
+      state,
+      action: PayloadAction<NavigatorState['narratives']>
+    ) => {
+      if (state.fresh) return;
+      state.fresh = true;
+      state.narratives = action.payload;
+    },
```
[the remark this answers is not in the document]
Comment (dakotablair) [reply src/features/navigator/navigatorSlice.ts:47]: There's an explanation in the git commit message:
```
This introduces the navigator.fresh state which is a boolean representing that
the items held in navigator.items are up to date with user input. When they are
out of date navigator.fresh is false. The refresh button provides the user with
the ability to set this for any reason, such as when a narrative is shared by
another user.
```
Should I include something like that in `navigatorSlice.ts` as a comment?

File: src/common/components/Table.tsx [hunk 3]
```diff
@@ -33,7 +33,7 @@ export type ColumnDefs<Datum> = (
  * [TanStack/table](https://tanstack.com/table/v8/docs/) see those docs for
  * detailed documentation on column definitions.
  */
-export const Table = <Datum extends unknown>({
+export const Table = <Datum,>({
```
[the remark this answers is not in the document]
Comment (dakotablair) [reply src/common/components/Table.tsx:36]: Yep! 🤷

File: src/features/navigator/navigatorSlice.ts [hunk 4]
```diff
@@ -20,18 +29,50 @@ export const navigatorSlice = createSlice({
...
+export const useNarratives = (query?: string) => {
+  const dispatch = useAppDispatch();
+  const result = queryNarratives(query);
+  const narrativesPrevious = useAppSelector(narratives);
+  const fresh = useAppSelector(isFresh);
+  useEffect(() => {
+    dispatch(setNarratives(result));
+  }, [dispatch, fresh, narrativesPrevious, result]);
+  return result;
+};
```
[the remark this answers is not in the document]
Comment (dakotablair) [reply src/features/navigator/navigatorSlice.ts:78]: Based on our discussion we should say, "[Goodbye, useEffect](https://www.youtube.com/watch?v=bGzanfKVFeU)" here and connect the search api with RTK Query (which will provide a function to execute the query again).

**Comment by dakotablair** [comment]
This PR was created solely to facilitate review. See [this comment](https://github.com/kbase/ui-refresh-test/pull/52#issuecomment-1361566084) on PR #52 to see next efforts.

**Pull request closed without merging**